pkg/algorithm/hived_algorithm.go

```diff
@@ -49,6 +47,8 @@ type HivedAlgorithm struct {
 	freeCellList map[CellChain]ChainCellList
 	// all affinity groups that have been allocated cells
 	allocatedAffinityGroups map[string]*AlgoAffinityGroup
+	// all affinity groups that are preempting other groups
+	preemptorAffinityGroups map[string]*AlgoAffinityGroup
```

Comment:
Merge to one slice, and use AffinityGroupState to judge?

Reply:
Merged to one structure, h.allocatedGroups

Reply:
Do we need rename the functions like createPreemptorAffinityGroups, delelePreemptorGroups, to Preempting?